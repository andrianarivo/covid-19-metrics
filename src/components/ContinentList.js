import PropTypes from 'prop-types';
import styled from 'styled-components';
import ContinentItem from './ContinentItem';

export default function ContinentList(props) {
  const { continents } = props;
  return (
    <Continents>
      {continents.map((item) => (
        <ContinentItem
          key={item.continent}
          continent={item.continent}
          active={item.active}
          updated={item.updated}
        />
      ))}
    </Continents>
  );
}

ContinentList.propTypes = {
  continents: PropTypes.arrayOf(Object).isRequired,
};

const Continents = styled.ul`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;
