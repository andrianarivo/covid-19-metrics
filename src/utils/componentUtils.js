import asia from '../images/asia.png';
import africa from '../images/africa.png';
import northAmerica from '../images/north_america.png';
import southAmerica from '../images/south_america.png';
import oceania from '../images/oceania.png';
import europe from '../images/europe.png';

const getImage = (continent) => {
  switch (continent.toLowerCase()) {
    case 'asia':
      return asia;
    case 'africa':
      return africa;
    case 'north america':
      return northAmerica;
    case 'south america':
      return southAmerica;
    case 'australia-oceania':
      return oceania;
    case 'europe':
      return europe;
    default:
      return asia;
  }
};

export default getImage;
