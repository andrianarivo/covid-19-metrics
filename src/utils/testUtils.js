import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

export function withRouter(children) {
  return (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );
}

export function withProvider(children, mockStore = store) {
  return (
    <Provider store={mockStore}>
      {children}
    </Provider>
  );
}
