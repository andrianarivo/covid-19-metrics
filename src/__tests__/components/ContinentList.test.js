import { create } from 'react-test-renderer';
import ContinentList from '../../components/ContinentList';
import { withRouter } from '../../utils/testUtils';

describe('Testing ContinentList component', () => {
  test('renders correctly', () => {
    const continentItems = [
      {
        updated: 1687952392990,
        active: 2389357,
        continent: 'North America',
      },
      {
        updated: 1687955393389,
        active: 14773130,
        continent: 'Asia',
      },
    ];
    const tree = create(withRouter(<ContinentList continents={continentItems} />));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
