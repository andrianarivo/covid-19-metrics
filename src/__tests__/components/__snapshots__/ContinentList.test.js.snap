// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContinentList component renders correctly 1`] = `
<ul
  className="sc-hVvQAm cwVPRp"
>
  <li
    className="sc-kIDzkD eneJoT"
    image="North America"
  >
    <p>
      North America
    </p>
    <p>
      2389357
    </p>
    <p>
      a day ago
    </p>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-arrow-right sc-jzuGiw knwkHW"
      data-icon="circle-arrow-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <button
      data-testid="button-North America"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </li>
  <li
    className="sc-kIDzkD jHRNaF"
    image="Asia"
  >
    <p>
      Asia
    </p>
    <p>
      14773130
    </p>
    <p>
      a day ago
    </p>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-arrow-right sc-jzuGiw knwkHW"
      data-icon="circle-arrow-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <button
      data-testid="button-Asia"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </li>
</ul>
`;
